Data Structures/Linked-Lists/Implementations of single linked list.cpp

```diff
@@ -0,0 +1,346 @@
+			printf("Enter the element before which to insert : ");
+			scanf("%d",&item);
+			start=addBefore(start,data,item);
+			break;
+		 case 9:
+			printf("Enter the element to be inserted : ");
+			scanf("%d",&data);
+			printf("Enter the position at which to insert : ");
+			scanf("%d",&pos);
+			start=addAtPos(start,data,pos);
+			break;
+		 case 10:
+			printf("Enter the element to be deleted : ");
+			scanf("%d",&data);
+			start=del(start, data);	
+			break;
+		 case 11:
+			start=reverse(start);
+			break;
+		 case 12:
+			 exit(1);
+		 default:
+			 printf("Wrong choice\n");
+		}/*End of switch */
+	}/*End of while */
+return 0;
+}
+/*End of main()*/
+
+struct node *createList(struct node *start)
+{
+	int i,n,data;
+	printf("Enter the number of nodes : ");
+	scanf("%d",&n);
+	start=NULL;
+	if(n==0)
+		return start;
+
+	printf("Enter the element to be inserted : ");
+	scanf("%d",&data);
+	start=addAtBeg(start,data);
+
+	for(i=2;i<=n;i++)
+	{
+		printf("Enter the element to be inserted : ");
+		scanf("%d",&data);
+		start=addAtEnd(start,data);	
+	}
+	return start;
+}
+/*End of create_list()*/
+
+void display(struct node *start)
+{
+	struct node *p;
+	if(start==NULL)
+	{
+		printf("List is empty\n");
+		return;
+	}
+	p=start;
+	printf("List is :\n");
+	while(p!=NULL)
+	{
+		printf("%d ",p->info);
+		p=p->link;
+	}
+	printf("\n\n");
+}
+/*End of display() */
+
+void count(struct node *start)
+{
+	struct node *p;
+	int cnt=0;
+	p=start;
+	while(p!=NULL)
+	{
+		p=p->link;
+		cnt++;
+	}
+	printf("Number of elements are %d\n",cnt);
+}
+/*End of count() */
+
+void search(struct node *start,int item)
+{
+	struct node *p=start;
+	int pos=1;
+	while(p!=NULL)
+	{
+		if(p->info==item)
+		{
+			printf("Item %d found at position %d\n",item,pos);
+			return;
+		}
+		p=p->link;
+		pos++;
+	}
+	printf("Item %d not found in list\n",item);
+}
+/*End of search()*/
+
+struct node *addAtBeg(struct node *start,int data)
+{
+	struct node *tmp;
+	tmp=(struct node *)malloc(sizeof(struct node));
+	tmp->info=data;
+	tmp->link=start;
+	start=tmp;
+	return start;
+}
+/*End of addAtBeg()*/
+
+struct node *addAtEnd(struct node *start,int data)
+{
+	struct node *p,*tmp;
+	tmp=(struct node *)malloc(sizeof(struct node));
+	tmp->info=data;
+	p=start;
+	while(p->link!=NULL)
+		p=p->link;
+	p->link=tmp;
+	tmp->link=NULL;
+	return start;
+}
+/*End of addAtEnd()*/
+
+struct node *addAfter(struct node *start,int data,int item)
+{
+	struct node *tmp,*p;
+	p=start;
+	while(p!=NULL)
+	{
+		if(p->info==item)
+		{
+			tmp=(struct node *)malloc(sizeof(struct node));
+			tmp->info=data;
+			tmp->link=p->link;
+			p->link=tmp;
+			return start;
+		}
+		p=p->link;
+	}
+	printf("%d not present in the list\n",item);
+	return start;
+}
+/*End of addafter()*/
+
+struct node *addBefore(struct node *start,int data,int item)
+{
+	struct node *tmp,*p;
+	if(start==NULL )
+	{
+		printf("List is empty\n");
+		return start;
+	}	
+	/*If data to be inserted before first node*/
+	if(item==start->info)

```

Comment:
You can directly call the **addAtBeg** function here for reusability of the code.